app/tests/cases_tests/test_api.py

```diff
@@ -243,20 +274,45 @@ def test_session_with_user_upload(client, obj, factory):
         creator=user,
     )
 
-    response = get_view_for_user(
-        viewname="api:upload-session-list",
-        user=user,
-        client=client,
-        method=client.post,
-        content_type="application/json",
-        data={"uploads": [upload.api_url], obj: o.slug},
-        HTTP_X_FORWARDED_PROTO="https",
-    )
-
-    assert response.status_code == 201
-    upload_session = response.json()
-
-    assert upload_session["uploads"] == [upload.api_url]
+    if interface:
+        data = {
+            "uploads": [upload.api_url],
+            obj: o.slug,
+            "interface": interface,
+        }
+    else:
+        data = {"uploads": [upload.api_url], obj: o.slug}
+
+    with capture_on_commit_callbacks(execute=True):
+        response = get_view_for_user(
+            viewname="api:upload-session-list",
+            user=user,
+            client=client,
+            method=client.post,
+            content_type="application/json",
+            data=data,
+            HTTP_X_FORWARDED_PROTO="https",
+        )
+
+    if obj == "reader_study" and interface:
```

Comment:
Just a general comment (and this may be hypocritical as I know we do this in the tests already, but, I want to try and improve the tests as we go). I think that it is best to not have any logic in the tests, if you need it, write a new test (use the assemble, act, assert pattern). For test code I think that it is better to duplicate as if tests get too complex they are hard to maintain, and it can be difficult to understand the coders intent.